Many small changes from runthrough

tbekolay opened the pull request:
In going through the Python 3 runthrough (see #153) I saw a bunch of little things that I felt should be changed.

Note that this branch is based off of #153 (i.e., it picks up where that PR leaves off -- it assumes that that PR has already been merged) so I recommend that we deal with #153 first, and even merge `py3k` into `gh-pages` before looking at this PR (it'll make merging much simpler).

I also think that we should ensure that these fixes aren't already done in other PRs. I was just blindly going through and changing things I saw that needed changing, but some of them struck me as things I've seen fly through my email at one time or another, and I definitely would rather an earlier PR be merged before looking at this one.

The history of this PR will be easier to read once those other things are merged, but if you want to check out the changes now, look at the Commits tab. This PR is essentially all of the commits from 20923d8c9a5770a9 onwards.

tbekolay commented:
With #153 and #155 merged, this is now ready for review. I've rebased it, so only the appropriate commits show up.

abostroem commented on 05-cond.md:
```diff
@@ -86,7 +86,7 @@ We can also combine tests using `and` and `or`.
 
 ~~~ {.python}
-if (1 > 0) and (-1 > 0):
+if 1 > 0 and -1 > 0:
```
@tbekolay why did you remove the parentheses? I think they improve readability even if they are not strictly necessary.

tbekolay replied:
Pedantically, I think that we should avoid unnecessary operators because we're implicitly teaching that they *are* necessary. Practically, there are many languages that require parentheses in if statements (`if (expression)`) and so I want to reinforce that this isn't necessary  (and, to me, looks really weird) in Python.

But, I don't feel that strongly about it so I'm happy to add them back in; anyone case to break the tie here?

tpoisot replied:
I've seen people get confused when there are no parentheses: `1 > 0 and -1` can be parsed as `1 > (0 and -1)` -- unless you've already read the whole line. I tend to use parentheses because they improve *reading* code, which I think is worth emphasizing.

If we want to not use the parentheses, maybe there needs to be something on the precedence of operators, and I think this lesson is dense already.

I think the parentheses should stay.

abostroem commented on 05-cond.md:
```diff
@@ -98,7 +98,7 @@ one part is not true
 
 ~~~ {.python}
-if (1 < 0) or (-1 < 0):
+if 1 < 0 or -1 < 0:
```
Same as line 89

abostroem commented on instructors.md:
```diff
@@ -529,36 +551,35 @@ Solutions to exercises:
-> The third assertion checks that the running total always increases.
->   Input such as `[0,1,3,-5,4]` will make it fail.
-> ~~~
+>
```
I think we were using .output to indicate that this was the solution. Do you think we need some kind of visual cue to separate the question from the answer?

tbekolay replied:
I think my preference would probably be to remove the question, just to reduce duplication, as when we make minor changes to the questions we're unlikely to update them here as well. Mainly, I changed it because the `{.output}` also makes it a `pre` block, so things like the backticks here don't make a difference, and everything is rendered in a monospaced font.

I think ideally, we would implement a `{.answer}` style, which could even hide the answer until a click or something like that. For the time being, I'll add in a horizontal rule to at least give some visual clue that the answer's started.

abostroem commented on instructors.md:
```diff
@@ -570,51 +591,50 @@ Solutions to exercises:
+> ~~~ {.python}
 > # this is code/arith.py
->import sys
+> import sys
```
Why the space in front of sys?

tbekolay replied:
I always thought the space between the `>` and the rest of the line was necessary in Markdown? It's certainly [there in the spec](https://daringfireball.net/projects/markdown/syntax#blockquote). I suspect if the first line didn't have the extra space, the formatting would be all messed up... but also, it's inconsistent to have a space in the first line but not subsequent lines.

abostroem commented on instructors.md:
```diff
@@ -570,51 +591,50 @@ Solutions to exercises:
 >
->def main():
->    assert len(sys.argv) == 4, 'Need exactly 3 arguments'
+> def main():
```
again, why the space?

abostroem commented on instructors.md:
```diff
@@ -570,51 +591,50 @@ Solutions to exercises:
+>     do_arithmetic(operand1, operator, operand2)
 >
->def do_arithmetic(operand1, operator, operand2):
+> def do_arithmetic(operand1, operator, operand2):
```
Space?

abostroem commented on instructors.md:
```diff
@@ -570,51 +591,50 @@ Solutions to exercises:
+>     print(value)
 >
->main()
+> main()
```
space?

abostroem commented on instructors.md:
```diff
@@ -631,63 +651,62 @@ Solutions to exercises:
+> import glob
 >
->main()
+> def main():
```
space?

abostroem commented:
@tbekolay these changes mostly look good (and will clean up the look of the page considerably). A few minor inline comments then we can merge.

tbekolay commented:
I removed the contentious change with the parentheses around if conditions, but left in the spaces in the blockquotes because it's consistent with how we do it everywhere else in the lesson. I added horizontal lines to separate the question from the answer in the instructor guide (see 41e24ef320824a249bd0c20e). The rest is as it was before. Would be good to get this merged soon, as we're starting to get an influx of PRs that are duplicating this effort.

abostroem commented:
I merged this, but when I run make preview  and do a git status only 07, 10, and discussion show up as different. Any ideas what's going on? (if you have the html generated feel free to push them)

tbekolay commented:
Not sure how make determines newness, but you can do `make --always-make preview` to force it to regenerate all of them. Thanks for merging :)
